arm-machinelearning/2016-05-01-preview/swagger/webservices.json

```diff
@@ -0,0 +1,915 @@
+          }
+        }
+      }
+    }    
+  },
+  "parameters": {
+    "SubscriptionIdParameter": {
+      "name": "subscriptionId",
+      "description": "Azure subscription id.",
+      "in": "path",
+      "type": "string",
+      "required": true
+    },
+    "APIVersionParameter": {
+      "in": "query",
+      "name": "api-version",
+      "type": "string",
+      "description": "The versiong of the Microsoft.MachineLearning resource provider API to be used.",
+      "required": true
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "required": [
+        "location"
+      ],
+      "properties": {
+        "id": {
+          "description": "Resource Id",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource Name",
+          "type": "string"
+        },
+        "location": {
+          "description": "Resource Location",
+          "type": "string"
+        },
+        "type": {
+          "description": "Resource type",
+          "type": "string",
+          "readOnly": true
+        },
+        "tags": {
+          "description": "Resource tags",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "WebService": {
+      "type": "object",
+      "description": "Instance of an Azure ML web service resource.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/WebServiceProperties",
+          "description": "Web service resource properties."
+        }
+      }
+    },
+    "WebServiceProperties": {
+      "type": "object",
+      "description": "The set of properties specific to the Azure ML web service resource.",
+      "required": [
+        "packageType"
+      ],
+      "discriminator": "packageType",
+      "properties": {
+        "title": {
+          "type": "string",
+          "description": "The title of the Azure ML web service."
+        },
+        "description": {
+          "type": "string",
+          "description": "The description of the Azure ML web service."
+        },
+        "createdOn": {
+          "type": "string",
+          "description": "The moment of time the Azure ML web service was created.",
+          "format": "date-time",
+          "readOnly": true
+        },
+        "modifiedOn": {
+          "type": "string",
+          "description": "The moment of time the web service was last modified.",
+          "format": "date-time",
+          "readOnly": true
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "The web service resource's provisioning state.",
+          "enum": [
+            "Unknown",
+            "Provisioning",
+            "Succeeded",
+            "Failed",
+            "Canceled"
+          ],
+          "readOnly": true         
```

Comment:
That is fine. This can be achieved by setting "modelAsString" ptoperty of the `x-ms-enum` extension to true. That means, AutoRest will not generate an enum. In Languages that support Enum, it will generate a static class with properties. This gives users better intellisense. In languages that do no support Enum it will be treated as a string.

If you read the documentation link in the previous comment, you will get a better idea about the extension